salt/modules/virt.py

```diff
@@ -273,21 +274,28 @@ def _gen_xml(name,
         else:
             context['telnet_port'] = 23023  # FIXME: use random unused port
     if 'serial_type' in context:
+        if context['serial_type'] == 'nmdm' and 'nmdm_id' in kwargs:
+            context['nmdm_id'] = kwargs['nmdm_id']
```

Comment:
It seems to me that this function is getting a lot of kwargs now, most of them undocumented. I wonder if we could come up with something cleaner than just adding kwargs every now and then.

Reply:
That comment still applies: ``nmdm_id`` should be moved into a normal parameter with default value rather than a ``**kwargs``.

Reply:
Now bhyve has good support for VNC I'd suggest to omit the nmdm serial type.